### ISSUE TITLE
Real-time metadata creation not working with lookups

### ISSUE THREAD

**Issue by freiman** [OpenTSDB/opentsdb#900]
We have enabled real-time metadata creation.

Here is actual configuration of opentsdb:
`{"tsd.core.auto_create_metrics":"true","tsd.core.auto_create_tagks":"true","tsd.core.auto_create_tagvs":"true","tsd.core.connections.limit":"0","tsd.core.enable_api":"true","tsd.core.enable_ui":"true","tsd.core.meta.cache.enable":"false","tsd.core.meta.enable_realtime_ts":"true","tsd.core.meta.enable_realtime_uid":"true","tsd.core.meta.enable_tsuid_incrementing":"false","tsd.core.meta.enable_tsuid_tracking":"false","tsd.core.plugin_path":"/usr/share/opentsdb/plugins","tsd.core.preload_uid_cache":"false","tsd.core.preload_uid_cache.max_entries":"300000","tsd.core.socket.timeout":"0","tsd.core.stats_with_port":"false","tsd.core.storage_exception_handler.enable":"false","tsd.core.tree.enable_processing":"false","tsd.core.uid.random_metrics":"false","tsd.http.cachedir":"/tmp/opentsdb","tsd.http.query.allow_delete":"false","tsd.http.request.cors_domains":"*","tsd.http.request.cors_headers":"Content-Type","tsd.http.request.enable_chunked":"false","tsd.http.request.max_chunk":"4096","tsd.http.show_stack_trace":"true","tsd.http.staticroot":"/usr/share/opentsdb/static/","tsd.mode":"rw","tsd.network.async_io":"true","tsd.network.bind":"0.0.0.0","tsd.network.keep_alive":"true","tsd.network.port":"4242","tsd.network.reuse_address":"true","tsd.network.tcp_no_delay":"true","tsd.network.worker_threads":"","tsd.no_diediedie":"false","tsd.query.allow_simultaneous_duplicates":"true","tsd.query.enable_fuzzy_filter":"true","tsd.query.filter.expansion_limit":"4096","tsd.query.skip_unresolved_tagvs":"false","tsd.query.timeout":"0","tsd.rtpublisher.enable":"false","tsd.rtpublisher.plugin":"","tsd.search.enable":"false","tsd.search.plugin":"","tsd.startup.enable":"false","tsd.startup.plugin":"","tsd.stats.canonical":"false","tsd.storage.compaction.flush_interval":"10","tsd.storage.compaction.flush_speed":"2","tsd.storage.compaction.max_concurrent_flushes":"10000","tsd.storage.compaction.min_flush_threshold":"100","tsd.storage.enable_appends":"false","tsd.storage.enable_compaction":"true","tsd.storage.fix_duplicates":"true","tsd.storage.flush_interval":"1000","tsd.storage.hbase.data_table":"tsdb","tsd.storage.hbase.meta_table":"tsdb-meta","tsd.storage.hbase.prefetch_meta":"false","tsd.storage.hbase.scanner.maxNumRows":"128","tsd.storage.hbase.tree_table":"tsdb-tree","tsd.storage.hbase.uid_table":"tsdb-uid","tsd.storage.hbase.zk_basedir":"/hbase","tsd.storage.hbase.zk_quorum":"localhost","tsd.storage.repair_appends":"false","tsd.timeseriesfilter.enable":"false","tsd.uidfilter.enable":"false"}
`

I hope that "tsd.core.meta.enable_realtime_ts":"true" and "tsd.core.meta.enable_realtime_uid":"true" should enable real-time metadata creation, but problem is that /api/search/lookup endpoint not returning newly created metadata (metric, tagk, tagv).

We must run tsdb cli utility to sync metada:
`/usr/share/opentsdb/bin/tsdb uid metasync`

And after that new metadata entries are showing.

We have debian distibution and opentsdb is installed as debian package. I tested it on actual version 2.3.0-RC2, but it is not working also in older versions. HBase is in version 1.2.0+cdh5.9.0+205-1.cdh5.9.0.p0.30~wheezy-cdh5.9.0.

Can you fix this? Or how else could I get a list of metric with their tags?
Thank you

**Comment by manolama** [comment]
Hi @freiman, sorry for the confusion around this setting. You'll also need to set ``tsd.core.meta.enable_tsuid_incrementing`` to true for the real-time meta code to work. Give that a try please.

**Comment by freiman** [comment]
Hi @manolama, thank you! I tried this and it was better, but it wasn't the best. New metrics were available through /api/search/lookup endpoint after putting them into opentsdb, but not the tag keys and values.
But after that i tried tsd.core.meta.enable_tsuid_tracking = true and it helps. Now all metadata are available after putting them into opentsdb. Many thanks for pointing me to this.

You can close this issue. I have only one question. What is tsd.core.meta.cache.enable option? Thanks

**Comment by freiman** [comment]
Turning on these options increased the load on the server as it is written in the documentation. Is there any plan to separate enabling of realtime metadata synchronization from `tsd.core.meta.enable_tsuid_incrementing` and `tsd.core.meta.enable_tsuid_tracking`? Or maybe I dont understand meaning of these options.

**Comment by manolama** [comment]
Glad it worked, I'll close this and respond :)

The ``tsd.core.meta.cache.enable`` is for a plugin interface that helps to decrease the load on the HBase servers by spooling updates locally in the TSDs, then periodically updating the meta table. I didn't finish work on the plugin implementation as we went a different route for meta-data. The route we took forks data spooled in Kafka into a separate set of servers that deduplicate and update meta in a similar way. We hope to open source that code soon. It puts much less strain on the region servers.
